src/fsharp/FSharp.Core/list.fs

```diff
@@ -692,3 +692,51 @@ namespace Microsoft.FSharp.Collections
 
         [<CompiledName("Unfold")>]
         let unfold<'T, 'State> (generator:'State -> ('T*'State) option) (state:'State) = Microsoft.FSharp.Primitives.Basics.List.unfold generator state
+
+        [<CompiledName("RemoveAt")>]
+        let removeAt (index: int) (source: 'T list) : 'T list =
+            if index < 0 || index >= source.Length then invalidArg "index" "index must be within bounds of the array"
+
+            let before, tail = Microsoft.FSharp.Primitives.Basics.List.splitAt index source
+            let _, after = Microsoft.FSharp.Primitives.Basics.List.splitAt 1 tail
+            before @ after
```

Comment:
There will be a better implementation using an explicit [ListCollector](https://github.com/dotnet/fsharp/blob/f45ef819b5310de521645dbc68ccbfc57f8edda4/src/fsharp/FSharp.Core/seqcore.fsi#L161) something like this
```fsharp
            let mutable i = 0
            let mutable coll = ListCollector()
            let mutable curr = source
            while i < index do
                  match curr with
                  | [] ->invalidArg "index" "index must be within bounds of the list" 
                  | h::t ->
                      coll.Add(h)
                      curr <- t
                  i <- i + 1
            coll.AddManyAndClose(curr.Tail)
```

Reply:
I rewrote all the list implementations to use ListCollector instead